db/migrate/20210922130721_add_vendor_data_to_providers.rb

```diff
@@ -0,0 +1,8 @@
+class AddVendorDataToProviders < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  def change
+    add_reference :providers, :vendor, null: true, index: {algorithm: :concurrently}
+    add_column :providers, :vendor_product, :string
```

Comment:
Does this feel a bit odd – the vendor lives on a separate table, but the product lives on `Provider`?

Why not have the vendor/product together on one table? eg

```ruby
v = VendorIntegration.find_by(vendor: :ellucian, product: :banner) # enums — i18n can render these nicely when needed
Provider.where(vendor_integration: v)
```

And we'd retain the ability to select providers by vendor, even when products differed, eg

```ruby
all_ellucian_integrations = VendorIntegration.where(vendor: :ellucian)
all_ellucian_providers = Provider.where(vendor_integration: all_ellucian_integrations)
```

Reply:
Depends on what we are planning to do with 'product', I guess. My assumption so far is we're not going to use it (!) so another direction would be to just drop it. The Slack notifications we'll build will not be product-specific, as far as I know.

Reply:
OK — if we're not fussed about it I'd leave it out, then? And keep `Vendor`. Would you consider calling the vendor name `name` rather than `code`?

Reply:
Thought a bit more about this and discussed briefly with Abeer, I think it may be worth refactoring to use `VendorIntegration`, after all -- not for the upcoming slack notifications story but to allow us to troubleshoot integrations at the product level. This will turn `code` into just `vendor` :-)

As long as we can rely on product information for HEIs, of course. Will check.

Reply:
Turns out we cannot rely on the product information we hold over time. Removing completely.